camayoc/tests/qcs/api/v1/scans/test_scans_common.py

```diff
@@ -127,3 +126,44 @@ def test_scan_with_optional_products(scan_type, shared_client, cleanup):
         scan.create()
         cleanup.append(scan)
         assert scan.equivalent(scan.read().json())
+
+
+@pytest.mark.parametrize('scan_type', ['connect', 'inspect'])
+def test_negative_disable_optional_products(scan_type, shared_client, cleanup):
+    """Attempt to disable optional products with non-acceptable booleans.
+
+    :id: 2adb483c-578d-4131-b426-9f772c1803de
+    :description: Create a scan with bad input for optional products
+    :steps:
+        1) Create a network credential
+        2) Create network source using the network credential.
+        3) Create a scan using the network source. When creating the scan
+           disable the optional products with bad values.
+    :expectedresults: The scan is not created.
+    """
+    num_products = random.randint(1, len(QCS_OPTIONAL_PRODUCTS))
+    product_combinations = combinations(QCS_OPTIONAL_PRODUCTS, num_products)
+    for combo in product_combinations:
+        source_ids = []
+        for _ in range(random.randint(1, 10)):
+            src_type = random.choice(QCS_SOURCE_TYPES)
+            src = gen_valid_source(cleanup, src_type, 'localhost')
+            source_ids.append(src._id)
+        scan = Scan(
+            source_ids=source_ids,
+            scan_type=scan_type,
+            client=shared_client)
+        products = {p: True for p in QCS_OPTIONAL_PRODUCTS if p not in combo}
+        bad_choice = random.choice(['hamburger', '87', 42, '*'])
+        products.update({p: bad_choice for p in combo})
+        scan.options.update({OPTIONAL_PROD: products})
+        echo_client = api.Client(response_handler=api.echo_handler)
+        scan.client = echo_client
+        create_response = scan.create()
```

Comment:
Should there be a check for a `400` status code? Or is checking that it shouldn't be created good enough?

Reply:
Does not hurt! Better to check 400 to catch if it is a 500 for some reason, which we don't want. I've added it